src/constructs/core/ssm.ts

```diff
@@ -0,0 +1,97 @@
+import { readFileSync } from "fs";
+import { join } from "path";
+import type { IGrantable, IPrincipal } from "@aws-cdk/aws-iam";
+import { Policy, PolicyStatement } from "@aws-cdk/aws-iam";
+import { Code, Runtime, SingletonFunction } from "@aws-cdk/aws-lambda";
+import type { Reference } from "@aws-cdk/core";
+import { Construct, CustomResource, Duration } from "@aws-cdk/core";
+import { AwsCustomResourcePolicy } from "@aws-cdk/custom-resources";
+import type { CustomResourceGetParameterProps } from "./custom-resources/interfaces";
+import type { AppIdentity } from "./identity";
+import type { GuStack } from "./stack";
+
+export interface GuSSMParameterProps {
+  parameter: string;
+  secure?: boolean;
+}
+
+export interface GuSSMIdentityParameterProps extends GuSSMParameterProps, AppIdentity {}
+
+const stripped = (str: string) => str.replace(/[-/]/g, "");
+
+export class GuSSMParameter extends Construct implements IGrantable {
+  private readonly customResource: CustomResource;
+  readonly grantPrincipal: IPrincipal;
+
+  // eslint-disable-next-line custom-rules/valid-constructors -- TODO: Remove once linting rules have been relaxed for this
+  constructor(scope: GuStack, props: GuSSMParameterProps) {
+    const { parameter } = props;
+
+    const id = (id: string) => {
+      const now = Date.now();
+      // We need to create UIDs for the resources in this construct, as otherwise CFN will not trigger the lambda on updates for resources that appear to be the same
+      const uid = now.toString().substr(now.toString().length - 4);
+      return parameter.toUpperCase().includes("TOKEN") ? `${id}-token-${uid}` : `${id}-${stripped(parameter)}-${uid}`;
+    };
+
+    super(scope, id("GuSSMParameter"));
+
+    const provider = new SingletonFunction(scope, id("Provider"), {
+      code: Code.fromInline(readFileSync(join(__dirname, "/custom-resources/runtime/lambda.js")).toString()),
+      runtime: Runtime.NODEJS_12_X, // TODO: Ensure that the TS -> JS compile creates a NODE12-compliant file
```

Comment:
Node 12 is in [maintenance LTS](https://nodejs.org/en/about/releases/). AWS recently announced [support for node 14 in AWS Lambda](https://aws.amazon.com/blogs/compute/node-js-14-x-runtime-now-available-in-aws-lambda/).

How does the output for a node 14 runtime break? There might be some TS config tweaks we can make? Or raise it with our AWS TAM?

Reply:
I think this TODO is outdated, as the TS configuration in the app creates Node 12 compliant JS code. The problem with upgrading to Node 14 is that it for some reason prevents us from using inline code (ie `Code.fromInline`), which is necessary here. I can't find anything online yet that specifically explains why, but it means that we're somewhat restricted to 12 until we can inline the code or find a better solution.